### PR TITLE
Feature/mds 2665 create bond data structure

### PR DESCRIPTION
**Pull request by JustinMacaulay**
- Adds the base data structure to handle bonds.
- Also creates marsmallow schema for the bond structure.
- Adds the bond relationship to Permits.

### CODE REVIEW

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
+CREATE TABLE bond_status_code (
```
Comment (Jsyro) [inline migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: naming consistency i'm now keenly aware of, the table should be bond_status
and the pk should be bond_status_code

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    create_timestamp                 timestamp with time zone DEFAULT now() NOT NULL            ,
+    update_user                      varchar                                NOT NULL            ,
+    update_timestamp                 timestamp with time zone DEFAULT now() NOT NULL
+);
+
+ALTER TABLE bond_status_code OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS bond (
+    bond_id                                                          SERIAL PRIMARY KEY,
+    bond_guid                        uuid DEFAULT gen_random_uuid()            NOT NULL,
```
Comment (Jsyro) [inline migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: unique constraint for safety.

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    FOREIGN KEY (bond_status_code) REFERENCES bond_status_code(bond_status_code) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (bond_type_code) REFERENCES bond_type_code(bond_type_code) DEFERRABLE INITIALLY DEFERRED
+
+);
+
+ALTER TABLE bond OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS bond_permit_xref
+(
+    bond_permit_xref_guid uuid    DEFAULT gen_random_uuid() NOT NULL PRIMARY KEY,
```
Comment (exzizt) [inline migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: The XREF GUID isn't needed if we use a primary key of (bond_id, permit_id). I thought we wanted to ditch serial keys on XREF tables but maybe it doesn't matter so much anymore.

File: services/core-api/app/api/utils/static_data.py [hunk 4]
```diff
@@ -45,3 +45,5 @@ def setup_static_data(Base):
 
             except Exception as e:
                 raise e
+
+    current_app.logger.debug(STATIC_DATA)
```
Comment (exzizt) [inline services/core-api/app/api/utils/static_data.py]: Is this wanted debug?

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql (hunk 1 above)
Comment (JustinMacaulay) [reply migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: Im not saying one way or the other is right but we have a lot of type codes and status codes in our system named "x_status_code"  going forward are we aiming to change everything to the "bond_status" style?

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql (hunk 2 above)
Comment (JustinMacaulay) [reply migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: added

File: services/core-api/app/api/utils/static_data.py (hunk 4 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/utils/static_data.py]: removed.

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql (hunk 3 above)
Comment (JustinMacaulay) [reply migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: I don't remember that conversation or choice but am more than happy to.

File: migrations/sql/V2020.03.12.08.33__create_bond_structure.sql (hunk 1 above)
Comment (Jsyro) [reply migrations/sql/V2020.03.12.08.33__create_bond_structure.sql]: The code should refer to the identifier, each row in this table are a 'bond_status' the code is the identifier, i just looked at all our code tables (as you did too), and this inconsistency stood out.

**Pull request merged**